Fix: use mop for describe-object method

kilianmh opened the pull request:
A function `generate-method-descriptions` has been added. This function uses the mop for generating the method descriptions at run-time. Previously this happened at compile-time. The `generate-method-descriptions` is called from the generated specialized `describe-object` method. Now the output of `describe-object` should be correct again and show all generated, excluding the `describe-object` method itself.

Also type declarations have been added to changed functions: `generate-method-descriptions`, `generate-client-class`, `%generate-client`.

Closes #12

kilianmh committed:
Fix: use mop for describe-object method

A function `generate-method-descriptions` has been added. This function uses the mop
for generating the method descriptions at run-time. Previously this happend at compile-time.
The `generate-method-descriptions` is called from the generated specialized `describe-object`
method. Now the output of `describe-object` should be correct again and show all generated,
excluding the describe-object method itself.

Also type declarations have been added to changed functions: generate-method-descriptions,
generate-client-class, %generate-client.

svetlyak40wt commented on client/core.lisp:
```diff
@@ -25,27 +25,57 @@
-                       lambda-list))))
+  (declaim (ftype (function (closer-mop:specializer stream) null)
+		  generate-method-descriptions))
+  (defun generate-method-descriptions (class stream)
```
You don't need to wrap `generate-method-description` with `eval-when` anymore, because this function is not called during the macro-expansion.

kilianmh replied:
Correct. Now it is changed accordingly.

kilianmh committed:
Refactor: remove function from eval-when wrapper

The `generate-method-description` function is not called during
macroexpansion. Therefore no need to wrap it in eval-when.

svetlyak40wt commented:
Cool! Thank you!
